src/lib/components/Map/map.tsx

```diff
@@ -86,45 +75,41 @@ export const Map: React.FC<MapPropsType> = (props: MapPropsType): JSX.Element =>
-                setMapScale(newMapScale);
-                setViewCenterPoint(centerPoint);
-                setMinimapCenterPoint({
-                    x: (boundaryBox.width / 2) * newMapScale,
-                    y: (boundaryBox.height / 2) * newMapScale,
-                });
-            } else if (action === MapActionType.CenterView) {
-                const newMapScale = Math.min(
-                    Math.min(
-                        Math.max(
-                            Math.floor(
-                                Math.min(size.height / boundaryBox.height, size.width / boundaryBox.width) / 0.1
-                            ) *
-                                0.1 -
-                                0.1,
-                            0.5
-                        ),
-                        mapScale
-                    )
-                );
-                setMapScale(newMapScale);
-                setViewCenterPoint({
-                    x: (boundaryBox.width / 2) * newMapScale,
-                    y: (boundaryBox.height / 2) * newMapScale,
-                });
-                setMinimapCenterPoint({
-                    x: (boundaryBox.width / 2) * newMapScale,
-                    y: (boundaryBox.height / 2) * newMapScale,
-                });
-            }
+            const actions: { [key: string]: () => void } = {
+                [MapActionType.ZoomIn]: () => {
+                    const newScale = Math.min(3, scale + 0.1);
+                    setNewScale(newScale);
+                    setViewCenterPoint(centerPoint);
+                    setMinimapCenterPoint(centerPoint);
+                },
+                [MapActionType.ZoomOut]: () => {
+                    const newScale = Math.max(0.5, scale - 0.1);
+                    setNewScale(newScale);
+                    setViewCenterPoint(centerPoint);
+                    setMinimapCenterPoint({
+                        x: (boundaryBox.width / 2) * newScale,
+                        y: (boundaryBox.height / 2) * newScale,
+                    });
+                },
+                [MapActionType.CenterView]: () => {
+                    const mapSceneRatio = Math.min(size.height / boundaryBox.height, size.width / boundaryBox.width);
+                    const mapSceneRatioAdjustedTo10Percent = Math.floor(mapSceneRatio / 0.1) * 0.1;
+                    const mapSceneRatioWithMargin = mapSceneRatioAdjustedTo10Percent - 0.1;
+                    const newScale = Math.min(Math.max(mapSceneRatioWithMargin, 0.5), scale);
+                    setNewScale(newScale);
+                    setViewCenterPoint({
+                        x: boundaryBox.width / 2,
+                        y: boundaryBox.height / 2,
+                    });
+                    setMinimapCenterPoint({
+                        x: boundaryBox.width / 2,
+                        y: boundaryBox.height / 2,
+                    });
+                },
+            };
+            actions[action]();
```

Comment:
This is what I meant, but that is just my preference as I find it easier to read. Pick the one you prefer!
```typescript
switch (action) {
  case MapActionType.ZoomIn: {
    const newScale = Math.min(3, scale + 0.1);
    setNewScale(newScale);
    setViewCenterPoint(centerPoint);
    setMinimapCenterPoint(centerPoint);
    break;
  } case MapActionType.ZoomOut: {
    const newScale = Math.max(0.5, scale - 0.1);
    setNewScale(newScale);
    setViewCenterPoint(centerPoint);
    setMinimapCenterPoint({
      x: (boundaryBox.width / 2) * newScale,
      y: (boundaryBox.height / 2) * newScale,
    });
    break;
  } case MapActionType.CenterView: {
    const mapSceneRatio = Math.min(size.height / boundaryBox.height, size.width / boundaryBox.width);
    const mapSceneRatioAdjustedTo10Percent = Math.floor(mapSceneRatio / 0.1) * 0.1;
    const mapSceneRatioWithMargin = mapSceneRatioAdjustedTo10Percent - 0.1;
    const newScale = Math.min(Math.max(mapSceneRatioWithMargin, 0.5), scale);
    setNewScale(newScale);
    setViewCenterPoint({
      x: boundaryBox.width / 2,
      y: boundaryBox.height / 2,
    });
    setMinimapCenterPoint({
      x: boundaryBox.width / 2,
      y: boundaryBox.height / 2,
    });
  }
}
```

Reply:
I knew what you mean ;) But if you're fine with it I'd like to keep the one I have implemented now (object literal lookups) since switch statements use more lines of code and are more error prone (break).